core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -87,55 +107,212 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
   def map[B](f: A => B)(implicit F: Functor[F]): OptionT[F, B] =
     OptionT(F.map(value)(_.map(f)))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT.some[List](99)
+   *  scala> optionT.imap[Char](_.toChar)(_.toInt)
+   *  res0: OptionT[List, Char] = OptionT(List(Some(c)))
+   * }}}
+   */
   def imap[B](f: A => B)(g: B => A)(implicit F: Invariant[F]): OptionT[F, B] =
     OptionT {
       F.imap(value)(_.map(f))(_.map(g))
     }
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> type ToString[A] = Function1[A, String]
+   *  scala> val optionIntToString: ToString[Option[Int]] = oi => oi.map(_.toString + "!").getOrElse("default")
+   *  scala> val optionT: OptionT[ToString, Int] = OptionT[ToString, Int](optionIntToString)
+   *  scala> optionT.contramap[Double](_.toInt).value(Some(5.4321))
+   *  res0: String = 5!
+   * }}}
+   */
   def contramap[B](f: B => A)(implicit F: Contravariant[F]): OptionT[F, B] =
     OptionT {
       F.contramap(value)(_.map(f))
     }
 
   /**
    * Modify the context `F` using transformation `f`.
+   * 
+   * Example:
+   * {{{
+   *  scala> import cats.~>
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   * 
+   *  scala> val optionToList: Option ~> List = new ~>[Option, List] { override def apply[A](o: Option[A]): List[A] = o.toList }
+   *  scala> val optionT: OptionT[Option, Int] = OptionT.some(42)
+   *  scala> optionT.mapK[List](optionToList)
+   *  res0: OptionT[List, Int] = OptionT(List(Some(42)))
+   * }}}
    */
   def mapK[G[_]](f: F ~> G): OptionT[G, A] = OptionT[G, A](f(value))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(3), Some(5)))
+   *  scala> optionT.semiflatMap(x => List(x, x * 6))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(3), Some(18), Some(5), Some(30)))
+   * }}}
+   */
   def semiflatMap[B](f: A => F[B])(implicit F: Monad[F]): OptionT[F, B] =
     flatMap(a => OptionT.liftF(f(a)))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *  scala> import scala.util.Try
+   *
+   *  scala> // prints "5"
```

Comment:
Is there a good way to test if the side-effects worked properly?
If not, should we even leave those examples in?

Reply:
No, and `Try` is not sufficient for suspending side-effects anyway (nothing in Cats is). We should change these examples.

Reply:
Actually, `OptionT` does not have to deal with side effects:
```
scala> import cats.data.OptionT, cats.syntax.all._
import cats.data.OptionT
import cats.syntax.all._

scala> val optt = OptionT.some[Either[String, *]](3)
val optt: cats.data.OptionT[[β$0$]scala.util.Either[String,β$0$],Int] = OptionT(Right(Some(3)))

scala> optt.semiflatTap { case 1 | 2 | 3 => Right("hit!"); case _ => Left("miss!") }
val res0: cats.data.OptionT[[β$0$]scala.util.Either[String,β$0$],Int] = OptionT(Right(Some(3)))

scala> optt.semiflatTap { case 0 | 1 | 2 => Right("hit!"); case _ => Left("miss!") }
val res1: cats.data.OptionT[[β$0$]scala.util.Either[String,β$0$],Int] = OptionT(Left(miss!))

```

Reply:
That's very clever, I added it to the PR! :+1: